src/libraries/System.Diagnostics.Tracing/tests/BasicEventSourceTest/LoudListener.cs

```diff
@@ -18,7 +18,7 @@ internal class LoudListener : EventListener
 
         public LoudListener(EventSource eventSource)
         {
-            EnableEvents(eventSource, EventLevel.LogAlways, (EventKeywords)0xffffffff);
```

Comment:
I'm confused.  I thought `EnableEvents(eventSource, EventLevel.LogAlways, (EventKeywords)0xffffffff);` was basically saying "enable everything"... is it not?

Reply:
If that's not the case, or if this is making that not be the case, I fear the fallout will be huge.  Just in dotnet/runtime, don't all of these assume the same thing?
https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/System.Diagnostics.DiagnosticSource/tests/DiagnosticSourceEventSourceBridgeTests.cs#L1525-L1529
https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/Common/tests/System/Diagnostics/Tracing/ConsoleEventListener.cs#L22
https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/System.Net.Http/tests/StressTests/HttpStress/HttpEventListener.cs#L43
https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/TraceLogging/XplatEventLogger.cs#L167

Reply:
There is a bunch of confusion around this, and I am trying to get a better picture of the implications. For the moment the change is on hold.

Different people think it means different things. The rule that every other level follows is that events get logged when level_of_event <= level_of_filter. Since LogAlways = 0, that implies it is the most restrictive filter. Our docs also agree with that interpretation. However on the flip side we've got code examples like the ones you found that assume LogAlways is a special case value that is the most permissive filter and certainly other code in the wild could make the same assumptions. I am chatting on email with the ETW folks because there is also evidence they treat the filter with LogAlways as a special case.

Another possible outcome is that we go change all our docs to say that LogAlways is a special case. I'm trying to hunt down other places in code we've made this check to see if they consistently special case it though I am fearful we are probably applying inconsistent rules in different code paths.